mmv1/products/dialogflowcx/SecuritySettings.yaml

```diff
@@ -0,0 +1,161 @@
+      Note: deidentifyTemplate must be located in the same region as the SecuritySettings.
+      Format: projects/<Project ID>/locations/<Location ID>/deidentifyTemplates/<Template ID> OR organizations/<Organization ID>/locations/<Location ID>/deidentifyTemplates/<Template ID>
+  - !ruby/object:Api::Type::Array
+    name: 'purgeDataTypes'
+    description: |
+      List of types of data to remove when retention settings triggers purge.
+    item_type: !ruby/object:Api::Type::Enum
+      name: 'purgeDataType'
+      description: |
+        Type of data we purge after retention settings triggers purge.
+        * DIALOGFLOW_HISTORY: Dialogflow history. This does not include Cloud logging, which is owned by the user - not Dialogflow.
+      values:
+        - :DIALOGFLOW_HISTORY
+  - !ruby/object:Api::Type::NestedObject
+    name: 'audioExportSettings'
+    description: |
+      Controls audio export settings for post-conversation analytics when ingesting audio to conversations.
+      If retention_strategy is set to REMOVE_AFTER_CONVERSATION or gcs_bucket is empty, audio export is disabled.
+      If audio export is enabled, audio is recorded and saved to gcs_bucket, subject to retention policy of gcs_bucket.
+      This setting won't effect audio input for implicit sessions via [Sessions.DetectIntent](https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.sessions/detectIntent#google.cloud.dialogflow.cx.v3.Sessions.DetectIntent).
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'gcsBucket'
+        description: |
+          Cloud Storage bucket to export audio record to. Setting this field would grant the Storage Object Creator role to the Dialogflow Service Agent. API caller that tries to modify this field should have the permission of storage.buckets.setIamPolicy.
+      - !ruby/object:Api::Type::String
+        name: 'audioExportPattern'
+        description: |
+          Filename pattern for exported audio.
+      - !ruby/object:Api::Type::Boolean
+        name: 'enableAudioRedaction'
+        description: |
+          Enable audio redaction if it is true.
+      - !ruby/object:Api::Type::Enum
+        name: 'audioFormat'
+        description: |
+          File format for exported audio file. Currently only in telephony recordings.
+          * MULAW: G.711 mu-law PCM with 8kHz sample rate.
+          * MP3: MP3 file format.
+          * OGG: OGG Vorbis.
+        values:
+          - :MULAW
+          - :MP3
+          - :OGG
+  - !ruby/object:Api::Type::NestedObject
+    name: 'insightsExportSettings'
+    description: |
+      Controls conversation exporting settings to Insights after conversation is completed.
+      If retentionStrategy is set to REMOVE_AFTER_CONVERSATION, Insights export is disabled no matter what you configure here.
+    properties:
```

Comment:
nit: Consider making this subfield required, empty objects can act weird in Terraform. They'll still work- just act a little unusual.

Reply:
Fair enough, done.